std/meta.d

```diff
@@ -1313,6 +1314,45 @@ private template SmartAlias(T...)
 }
 
 /**
+  * This template instantiates a template by taking its parameters first, and
+  * the template last. This can be useful when using staticMap to apply various
+  * templates to the same set of parameters, since staticMap always appends the
+  * list of arguments to one template.
+  *
+  * `ApplyWith!(A0, A1, ..., An-1, An)` is equivalent to
+  * `An!(A0, A1, ..., An-1)`
+  *
+  * Params:
+  *    args = Arguments for instantiation, followed by template to instantiate.
+  * Returns:
+  *    Alias to instantiated template passed as last parameter, with other
+  *    parameters used for instantiation.
+  */
+template ApplyWith(Args...) if(Args.length >= 1)
+{
+    alias ApplyWith = Instantiate!(Args[$-1], Args[0 .. $-1]);
+}
+
+///
+@safe unittest
+{
+    import std.string: leftJustify, center, rightJustify;
+    alias functions = staticMap!(ApplyLeft!(ApplyWith, string),
+                                 leftJustify, center, rightJustify);
+    string result = "";
+    static foreach(f; functions)
```

Comment:
Spacing after `foreach` applies to static foreach too (that's why CircleCi is failing)